introduction-to-javascript-for-python-programmers.html

```diff
@@ -0,0 +1,376 @@
+
+1. JS is a braces and semicolons language while Python is white-space and indent/block based*
+2. Python doesn't require an implicit variable declaration such as `var`
+3. Javascript and Python are both dynamically typed, whereas javascript is weakly, python strongly typed.
+4. In python, `self` is explicitly passed to a member function, and is not a special keyword or anything. In javascript, `this` is dynamically scoped and you can fiddle with the scope of a member function by calling `apply()` on it.
+5. JavaScript is a very forgiving language. You rarely run into errors. `undefined` keyword takes care of all nuances. In Python, exceptions are thrown.
+6. ~~You can run JS on a browser but not Python~~ Checkout [Python in Browser with Iodide](https://iodide-project.github.io/iodide-examples/python.html) 🎉
+
+\*Semicolons are optional in JavaScript to a certain extent. Read [Guide to Semicolons in JS](https://news.codecademy.com/your-guide-to-semicolons-in-javascript/) for more information1
+
+%% md
+# Fundamentals
+
+The contents of this tutorial have been aggregated from various sources. You can check them out if you face any difficulty  - 
+
+1. [JavaScript Info](http://javascript.info/)
+2. [Mozilla MDN](https://developer.mozilla.org/bm/docs/Web/JavaScript)
+3. [W3Schools](https://www.w3schools.com/js/)
+
+## Printing and Logging
+
+Python and other languages have a `print` method to display corresponding constants and variables. JavaScript has been designed in a different way for the purpose of serving the web. Natively, we can either use `console.log` method which will print the result in the browser's console or use `alert` which will show a popup window.
+
+%% js
+console.log("Hello")
+
+alert("Hi there")
+
+%% md
+The result of `console.log` will appear in your browser console. To open your browser console you can do :
+
+In Chrome
+
+    Windows: Ctrl + Shift + J
+    Mac OS: Cmd + Opt + J
+
+In Firefox
+
+    Windows: Ctrl + Shift + K
+    Mac OS: Cmd + Opt + K
+
+Luckily, Iodide offers a method to print the values of variables so that you don't have to open your console again and again for seeing the result.
+
+%% js
+var myVar = "Awesome"
+myVar
+
+%% js
+// By the way this is how you comment
+
+/* 
+For multiline you can 
+use this
+*/
+
+%% md
+## Variables
+
+Unlike Python, in JavaScript we need to define variables before then can be used. JavaScript offers three keywords to define variables `var`, `let` and `const`.
+
+`var` - Doesn't have block scope. They are either global or function-wide.  
+`let` - Modern way of declaring variables with lexical scope.  
+`const` - Similar to `let` but value of the variable can't be changed once declared
+
+Below is an example which shows how `var` doesn't have a block scope.
+
+_PS. As Iodide uses `let` and `const` under the hood to evaluate these cells, using them in the notebook would throw an error._
```

Comment:
_using_ let and const  doesn't actually cause errors, but they don't escape the scope of "eval" block they are inside -- so you can use let and const _within_ a cell, but those variables will not be available in other cells. sometimes this can be useful if you want temporary values within a cell but you don't want to pollute your global namespace